django/template/base.py

```diff
@@ -1004,14 +1053,17 @@ def __repr__(self):
         return "<Variable Node: %s>" % self.filter_expression
 
     def render(self, context):
+        return ''.join(self.stream(context))
+
+    def stream(self, context):
         try:
             output = self.filter_expression.resolve(context)
         except UnicodeDecodeError:
             # Unicode conversion can fail sometimes for reasons out of our
             # control (e.g. exception rendering). In that case, we fail
             # quietly.
-            return ''
-        return render_value_in_context(output, context)
+            yield ''
+        yield render_value_in_context(output, context)
```

Comment:
Since the default behavior of ``stream`` is to yield the render value, would it be better to not modify nodes, like ``VariableNode`` and ``TextNode``, that only return a single value?